go/consensus/api/api.go

```diff
@@ -343,6 +340,24 @@ type Status struct { // nolint: maligned
 
 	// IsValidator returns whether the current node is part of the validator set.
 	IsValidator bool `json:"is_validator"`
+
+	// P2P is the P2P status of the node.
+	P2P *P2PStatus `json:"p2p,omitempty"`
+}
+
+// P2PStatus is the P2P status of a node.
+type P2PStatus struct {
+	// ID is the public key used for consensus P2P communication.
+	ID signature.PublicKey `json:"id"`
```

Comment:
Should we just call this a public key?
```suggestion
	// PubKey is the public key used for consensus P2P communication.
	PubKey signature.PublicKey `json:"pub_key"`
```

Reply:
We can, but in the registry it would still be `id` :smile: